[Choong Weng Sheng] iP

CVincent0907 opened the pull request:
# This is my pull request 
## Welcome to my pull request 
### Bello Welcome 
### This is my bullet list 
* Walao
* Seh
*  GGWP
### This is my number list 
1. Wan Tan Mee
2. Char Siu Fan
3. Siew Mai
### This is my fenced code block with syntax highlighting
```java
public static void main(String[]) {
    System.out.println("Hello World");
    System.exit(1);
}
```
### This is my task list 
- [x] Sleep
- [ ] Drink
### This is my emoji 
😄 
### This is my block quote 
> Hello I am Vincent
### This is my hyperlink
[STFW](www.google.com)
### This is my inline code 
Use `System.out.println("Hi")` in writing 
### This is bold word
**HAHAHAHA**
### This is italic word 
*Snake*
### This is a strikethrough
You ~CAN~ do this !

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

CVincent0907 committed:
Rename Duke.java to TearIT.java

CVincent0907 committed:
Rename, Greet, Exit

CVincent0907 committed:
Added an Echo file for looping function echo

CVincent0907 committed:
Added an echo looping function which gets user command and termintate when bye is received

CVincent0907 committed:
Modify the main function by adding in Echo's function to realize echoing

CVincent0907 committed:
Modify format by adding indentation

CVincent0907 committed:
Modify format by adding indentation

CVincent0907 committed:
Added list functionality and store functionality for every user input except bye and list in echo function

CVincent0907 committed:
Simplify echo function by adding in independent add function

CVincent0907 committed:
Modify Echo class by introducing Task object array and change the related part accordingly

CVincent0907 committed:
Added Task class to handle user input as Task object independently

CVincent0907 committed:
Added mark, unmark and overloaded list function for task checking purpose

CVincent0907 committed:
modify mark and unmark function

CVincent0907 committed:
Added a guarding condition so that unmark/ mark cannot exceed the existent taskcount and handle them with printed message

CVincent0907 committed:
Added Deadline and Todo class

CVincent0907 committed:
Override to String method and add in regex to extract deadline

CVincent0907 committed:
Override toString method and add in regex to extract from and to time alongside description

CVincent0907 committed:
Modify Task class to Abstract class and inherited by Event, Todo, Deadline class

CVincent0907 committed:
Override toString method and add in regex for extraction of todo description

CVincent0907 committed:
Modify add function to handle all the potential task which will call the newly added extractAndCreate function when dealing with todo, event, deadline case insensitively

CVincent0907 committed:
Formatted the toString function

CVincent0907 committed:
Add in error handling logic when dealing with NON Todo, Event, Deadline using boolean errorCode

CVincent0907 committed:
Added comment regarding the edge case that regex could fail

CVincent0907 committed:
Added error handling method for lousy command beside todo, deadline .... and also handle todo, deadline ... with empty arguement alongside with handling mark and umark when encountering invalid input format ..list error handling yet to be done

CVincent0907 committed:
Added a horizontal line after error catching

CVincent0907 committed:
Add reminder to find way to handle inability of regex in Event to handle format like 12/3/2028 with more than one / occurences

CVincent0907 committed:
Added semi automated testing testcases

CVincent0907 committed:
Added delete method for task

CVincent0907 committed:
Added reduceTaskCount method to aid the delete method in Echo.java

CVincent0907 committed:
Changes the structure of if else to switch statement and added error handling when dealing with users enter one or more spaces without a properinput

CVincent0907 committed:
Deleted some comments in echo function

CVincent0907 committed:
Refactor switch statement

CVincent0907 committed:
Add a TearIT.txt to store all the tasks input after

CVincent0907 committed:
Add extractTask, readFromFile and writeToFile methods to allow retrieval and storing of tasks into text format upon processing

CVincent0907 committed:
M
bran-					erge branch 'branch-Level-7'
Merge the task retrieval and storing methods in branch-Level-7 to master

CVincent0907 committed:
Refactor code

CVincent0907 committed:
Refactor regex in todo, deadline and event.java file and add dateTime object and formatter to hanlde in deadline and event.java

CVincent0907 committed:
Add in exception handling method for unintended deadline and event input and update regex in extraction of user and file input to create datetime object

CVincent0907 committed:
Merge branch 'branch-Level-8'
										Merge with branch-Level-8 to add in the updated code with dateTime feature implemented

CVincent0907 committed:
Update TearIT.txt to empty status

CVincent0907 committed:
Update the TearIT.txt file to empty status

CVincent0907 committed:
Add Parser.java, Storage.java, Ui.java, Tasklist.java to implement more OOP

CVincent0907 committed:
Update the regex to handle invalid dateTime in Deadline and Event.java

CVincent0907 committed:
Update the class with other newly added OOP classes

CVincent0907 committed:
Clear the TearIT.txt storage file

CVincent0907 committed:
Merge branch-Level-8 with newly added OOP to master branch

CVincent0907 committed:
Refactor Parser, Storage, Ui into application package, Todo, Deadline, Task, Tasklist, Event into task package

CVincent0907 committed:
Refactor the codes into packages

CVincent0907 committed:
Update the TearIT.txt to emtpy status

CVincent0907 committed:
Merge branch 'branch-Level-8'
Merge the package-refactored codes from branch-Level-8 to master branch

CVincent0907 committed:
Merge branch 'add-gradle-support' of https://github.com/CVincent0907/ip into add-gradle-support
Pull remote repo origin branch add-gradle-support to local branch
add-gradle-support

CVincent0907 committed:
Merge branch 'add-gradle-support'
Merge add-gradle-support branch with master branch to supply it with
gradle tools

CVincent0907 committed:
Configure the build.gradle to add in junit module

CVincent0907 committed:
Update the code in Parser.java and Storage.java to fulfill junit test requirement

CVincent0907 committed:
Add ParserTest.java file to accomodate junit tests for extractAndCreateTask and extractTaskFromFile methods in Parser

CVincent0907 committed:
Add TasklistTest.java file to accomodate junit tests for add and delete methods in Tasklist.java

CVincent0907 committed:
Set all the external libraries used by project to track

CVincent0907 committed:
Update build.gradle with in ShadowJar with jar file named TearIT

CVincent0907 committed:
Update Parser.java Storage.java and Ui.java so that the file path which is read from can be created when it does not exist instead of hardcoding it

CVincent0907 committed:
Add the data folder to store TearIT.text specifically outside of src folder

CVincent0907 committed:
Add javadocs to production classes and test classes including the non-trivial methods

CVincent0907 committed:
Merge branch 'branch-A-JavaDoc'
Merge codes updated with JavaDocs from branch-A-JavaDoc to branch master

CVincent0907 committed:
Update the codes with coding standard

CVincent0907 committed:
Merge branch-A-CodingStandard with updated code styles with branch master after resolving conflict

CVincent0907 committed:
Add in find method and list task method in Tasklist.java, getDescriptionLowerCase method in Task.java and use them to add in find functionality in Ui.java

CVincent0907 committed:
Merge branch-Level-9 with added find, list methods in Tasklist.java and getDescriptionLowerCase method in Task.java with master branch

CVincent0907 committed:
Update Task.java

Resolve conflict which I oversee.

CVincent0907 committed:
Pull from github

CVincent0907 committed:
Pull from origin and resolve conflicts

farhan-navas commented on lib/aopalliance-1.0.jar:
should these .jar files be committed? won't these files be generated automatically when you run gradle build? if you manually downloaded these thn nvm my comment ahaha

farhan-navas commented on src/main/java/task/Event.java:
```diff
@@ -0,0 +1,32 @@
+public class Event extends Task {
+
+    public static final String DATE_TIME_REGEX_1 =
+            "event\\s+(.*?)\\s+/from\\s+(0[1-9]|[12][0-9]|3[01])-(0[1-9]|1[0-2])-(\\d{4})\\s+" +
+                    "(00|01|02|03|04|05|06|07|08|09|10|11|12|13|14|15|16|17|18|19|20|21|22|23)([0-5][0-9])\\s" +
+                    "+/to\\s+(0[1-9]|[12][0-9]|3[01])-(0[1-9]|1[0-2])-(\\d{4})\\s+(00|01|02|03|04|05|06|07|08|09|10|11|12|13|14|15|16|17|18|19|20|21|22|23)([0-5][0-9])";
```
```suggestion
            "event\\s+(.*?)\\s+/from\\s+(0[1-9]|[12][0-9]|3[01])-(0[1-9]|1[0-2])-(\\d{4})\\s+"
            + "(00|01|02|03|04|05|06|07|08|09|10|11|12|13|14|15|16|17|18|19|20|21|22|23)([0-5][0-9])\\s"
            + "+/to\\s+(0[1-9]|[12][0-9]|3[01])-(0[1-9]|1[0-2])-(\\d{4})\\s+(00|01|02|03|04|05|06|07|08|09|10|11|12|13|14|15|16|17|18|19|20|21|22|23)([0-5][0-9])";
```

shouldn't the + symbol be on a new line to follow coding convention?

farhan-navas commented on src/main/java/task/Tasklist.java:
```diff
@@ -0,0 +1,302 @@
+     *     <li>A numbered list of tasks in the order they are stored.</li>
+     *     <li>A footer indicating the end of the task list.</li>
+     * </ul>
+     */
```
very comprehensive javadoc comments overall nice!

farhan-navas commented on src/main/java/TearIT.java:
```diff
@@ -0,0 +1,33 @@
+/**
+ * TearIT is a class that provides entry point to the user interaction simulator.
+ */
+public class TearIT {
```
maybe could be TearIt instead of TearIT to follow coding convention?

farhan-navas reviewed (commented):
Overall LGTM, not much to say great job!

CVincent0907 committed:
Add checkstyle configuration

CVincent0907 committed:
Add checkstyle configuration files

CVincent0907 committed:
Merge branch-A-Checkstyle with master branch

CVincent0907 committed:
Add resources directory to contain images and fxml

CVincent0907 committed:
Add controller elements under gui package

CVincent0907 committed:
Refactor the code to adopt javafx element

CVincent0907 committed:
Modify the test files after integration with fxml

CVincent0907 committed:
Add in fxml and jpeg elements

CVincent0907 committed:
Add in javafx dependencies into build.gradle

CVincent0907 committed:
Add in javafx dependencies into build.gradle

CVincent0907 committed:
Merge branch 'branch-Level-10'
Merge Level 10 changes to branch master

CVincent0907 committed:
Add in usage of varags via echo method

CVincent0907 committed:
Modify build.gradle

CVincent0907 committed:
Refactor codes

CVincent0907 committed:
Refactor code

CVincent0907 committed:
Add enableAssertions in build.gradle

CVincent0907 committed:
Add assertion to assume Ui.echo will not return empty strings regardless of input

CVincent0907 committed:
Update the TasklistTest.java

The TasklistTest.java has not been updated after Tasklist.java
is rectified at its print statements due to previous typo

This will make sure that the test cases will not be obsolete

Let's change the deadline xx /from dd-mm-yyyy hhmm /to dd-mm-yyyy hhmm
to event xx/ from dd-mm-yyyy hhmm /to dd-mm-yyyy hhmm

CVincent0907 committed:
Fix bug related to reading from file storage

The Ui.echo is called every time handleUserInput method is invoked.
However, the readFromStorage method is only invoked and stored into
the program when users input list.

The bug has to be fixed so that Ui.echo will invoke readFromStorage
method the first time Ui.echo is called regardless of the user input.

Let's move the code fragment with readFromStorage under the "list" case
to the beginning of the Ui.echo method.

CVincent0907 committed:
Refactor Tasklist.java and TearIT.java

CVincent0907 committed:
Refactor TearIT.java

CVincent0907 committed:
Add Command class to handle user command

CVincent0907 committed:
Refactor codes to make the methods shorter

CVincent0907 committed:
Merge pull request #1 from CVincent0907/branch-A-CodeQuality

Refactor codes to make methods shorter

CVincent0907 committed:
Merge branch 'master' of https://github.com/CVincent0907/ip
Pull remote master branch to local master branch with refactored codes

CVincent0907 committed:
Merge branch 'master' into branch-A-Assertions

CVincent0907 committed:
Merge pull request #2 from CVincent0907/branch-A-Assertions

Merge branch with assertions added to master branch

CVincent0907 committed:
Merge branch 'master' of https://github.com/CVincent0907/ip
Pull remote master branch to local master branch

CVincent0907 committed:
Modify TearIT.java

CVincent0907 committed:
Add Github Actions(CI/CD) tool in the form of gradle.yml file

CVincent0907 committed:
Modify gradle.yml

CVincent0907 committed:
Remove IO redirection job steps in gradle.yml

CVincent0907 committed:
Refactor codes

CVincent0907 committed:
Add ViewSchedules functionality

CVincent0907 committed:
Update the test cases in Tasklist.java

CVincent0907 committed:
Disable enableAssertions to avoid side effect during gradle run

CVincent0907 committed:
Add enable assertion will affect run comment

CVincent0907 committed:
Modify scheduleLookUp method into using stream

CVincent0907 committed:
Refactor codes

CVincent0907 committed:
Refactor code

CVincent0907 committed:
Refactor code

CVincent0907 committed:
Change output to systemResponse

CVincent0907 committed:
Tweak FXML file for better GUI

CVincent0907 committed:
Refactor code as suggested by bot

CVincent0907 committed:
Delete lib directory

CVincent0907 committed:
Change tearit to TearIT

CVincent0907 committed:
Merge branch 'master' of https://github.com/CVincent0907/ip
Pull from origin master branch

CVincent0907 committed:
Add in personality element chidori and rasengan interaction

CVincent0907 committed:
Ädd test cases

CVincent0907 committed:
Update README.md

CVincent0907 committed:
Update README.md

CVincent0907 committed:
Upload Ui snapshot

CVincent0907 committed:
Update README.md

CVincent0907 committed:
Update README.md

CVincent0907 committed:
Update README.md

CVincent0907 committed:
Refactor codes and add interactive commands

CVincent0907 committed:
Add in help feature

CVincent0907 committed:
Delete README.md outside the docs

CVincent0907 committed:
Update README.md
